src/Compilers/CSharp/Test/Semantic/Semantics/UsingDeclarationTests.cs

```diff
@@ -783,5 +783,52 @@ static void Main()
                 Diagnostic(ErrorCode.ERR_BadMemberFlag, "readonly").WithArguments("readonly").WithLocation(7, 22)
                 );
         }
+
+        [Fact]
+        [WorkItem(42362, "https://github.com/dotnet/roslyn/issues/42362")]
+        public void UsingDeclarationWithConstModifier()
+        {
+            var source = @"
+class Program
+{
+    static System.IDisposable F() => null;
+
+    static void Main()
+    {
+        using const System.IDisposable o = F();
```

Comment:
>const [](start = 14, length = 5)

Please test other modifiers that parser can put into the syntax tree: readonly, static, volatile, etc.